packages/catalog-process/src/services/catalogService.ts

```diff
@@ -560,12 +560,7 @@ export function uploadDocumentLogic({
   assertEServiceExist(eServiceId, eService);
   assertRequesterAllowed(eService.data.producerId, authData.organizationId);
 
-  const descriptor = eService.data.descriptors.find(
-    (d: Descriptor) => d.id === descriptorId
-  );
-  if (descriptor === undefined) {
-    throw eServiceDescriptorNotFound(eServiceId, descriptorId);
-  }
+  retrieveDescriptor(descriptorId, eService);
```

Comment:
Just a refactor. The `retrieveDescriptor` function was already implemented and used in some cases.